packages/model-viewer/src/test/features/scene-graph/texture-spec.ts

```diff
@@ -0,0 +1,153 @@
+
+      element.model!.materials[0]
+          .pbrMetallicRoughness.baseColorTexture!.setTexture(texture);
+
+      const threeTexture: ThreeTexture =
+          element.model!.materials[0]
+              .pbrMetallicRoughness.baseColorTexture?.texture
+              ?.source[$underlyingTexture]!;
+
+      expect(threeTexture.uuid).to.be.equal(newUUID);
+    });
+
+    test('Verify legacy correlatedObjects are updated.', async () => {
+      const element = new ModelViewerElement();
+      element.src = ASTRONAUT_GLB_PATH;
+      document.body.insertBefore(element, document.body.firstChild);
+      await waitForEvent(element, 'load');
+
+      const path = assetPath(
+          'models/glTF-Sample-Models/2.0/BoxTextured/glTF/CesiumLogoFlat.png');
+      const texture = await element.createTexture(path);
+      const newUUID: string|undefined =
+          texture?.source[$underlyingTexture]?.uuid;
+
+      element.model!.materials[0]
+          .pbrMetallicRoughness.baseColorTexture!.setTexture(texture);
+
+      const uuidFromTextureInfoObject: string =
+          (element.model!.materials[0]
+               .pbrMetallicRoughness.baseColorTexture?.[$correlatedObjects]
+               ?.values()
+               .next()
+               .value as ThreeTexture)
+              .uuid;
+      expect(uuidFromTextureInfoObject).to.be.equal(newUUID);
+
+      const uuidFromTextureObject: string =
+          (element.model!.materials[0]
+               .pbrMetallicRoughness.baseColorTexture?.[$correlatedObjects]
+               ?.values()
+               .next()
+               .value as ThreeTexture)
+              .uuid;
+      expect(uuidFromTextureObject).to.be.equal(newUUID);
+
+      const uuidFromSamplerObject: string =
+          (element.model!.materials[0]
+               .pbrMetallicRoughness.baseColorTexture?.texture
+               ?.sampler[$correlatedObjects]
+               ?.values()
+               .next()
+               .value as ThreeTexture)
+              .uuid;
+      expect(uuidFromSamplerObject).to.be.equal(newUUID);
+
+      const uuidFromImageObject: string =
+          (element.model!.materials[0]
+               .pbrMetallicRoughness.baseColorTexture?.texture
+               ?.source[$correlatedObjects]
+               ?.values()
+               .next()
+               .value as ThreeTexture)
+              .uuid;
+      expect(uuidFromImageObject).to.be.equal(newUUID);
+    });
+
+    test('Set a texture and then setURI', async () => {
+      const element = new ModelViewerElement();
+      element.src = ASTRONAUT_GLB_PATH;
+      document.body.insertBefore(element, document.body.firstChild);
+      await waitForEvent(element, 'load');
+
+      const texture = await element.createTexture(assetPath(
+          'models/glTF-Sample-Models/2.0/BoxTextured/glTF/CesiumLogoFlat.png'));
+
+      element.model!.materials[0]
+          .pbrMetallicRoughness.baseColorTexture!.setTexture(texture);
```

Comment:
Looks like the code to here is shared in all these tests. Let's pack it into a `beforeEach()` method to reduce duplication.

Reply:
Done.

Reply:
onBefore was not recognized, so I followed the pattern where it was being used elsewhere now getting things like:
```
Chrome Headless 91.0.4472.114 (Mac OS 10.15.7) ERROR
  Uncaught ReferenceError: describe is not defined
  at lib/test/features/scene-graph/texture-spec.js:22:1
 ```

Reply:
resolved with a helper method for now.

Reply:
Oh, sorry, I used the jasmine term instead of the chai term. Really wish we didn't have two different testing frameworks... You're looking for `setup` and `teardown`. If you look at some of the other MV tests you'll see examples. Remember to remove elements in `teardown` to keep the tests hermetic.